### PR TITLE
[#10122] Make relatedGiverEmail optional and hide moderate response button on null

### LINKED ISSUE #10122: Make relatedGiverEmail optional and fill it only if response giver is visible
The `relatedGiverEmail` is used for `Moderate Response` button, which takes us to the proper giver submission page.

For anonymous response, we shall not show `Moderate Response` button in the first place. In addition, the `relatedGiverEmail` also should be `null` in this case.

This will also solve the TODO: https://github.com/TEAMMATES/teammates/blob/master/src/main/java/teammates/ui/webapi/output/SessionResultsData.java#L267

Hint:
- Change frontend logic to show the button only if `relatedGiverEmail` is not null
- Follow the logic in #10107

### PR DESCRIPTION
**Pull request by Derek-Hardy**
Fixes #10122 

**Outline of Solution**
- add `@Nullable` to `relatedGiverEmail` field
- set `relatedGiverEmail ` to null when giver is not visible
- add users to relateGiverEmail map in `gqr` and `grq` view for existence check by `userInfo`
- include existence check in `ngIf` for `moderate response` button

### CODE REVIEW

File: src/main/java/teammates/ui/webapi/output/SessionResultsData.java [hunk 1]
```diff
@@ -271,6 +275,7 @@ public String getQuestionStatistics() {
          * Depending on the question giver type, {@code giverIdentifier} may contain the giver's email, any team member's
          * email or "anonymous".
```
Comment (xpdavid) [inline src/main/java/teammates/ui/webapi/output/SessionResultsData.java:276]: So we shall update the documentation also?

File: src/web/app/components/question-responses/per-question-view-responses/per-question-view-responses.component.html [hunk 2]
```diff
@@ -44,7 +44,7 @@
           <div *ngIf="response.participantComment" [innerHTML]="response.participantComment.commentText | safeHtml"></div>
         </td>
         <td>
-          <tm-response-moderation-button [session]="session" [relatedGiverEmail]="response.relatedGiverEmail" [moderatedQuestionId]="question.feedbackQuestionId"></tm-response-moderation-button>
+          <tm-response-moderation-button *ngIf="response.relatedGiverEmail" [session]="session" [relatedGiverEmail]="response.relatedGiverEmail" [moderatedQuestionId]="question.feedbackQuestionId"></tm-response-moderation-button>
```
Comment (xpdavid) [inline src/web/app/components/question-responses/per-question-view-responses/per-question-view-responses.component.html:47]: I think you also want to update `tm-response-moderation-button` component and remove the anonymous check.

File: src/web/app/pages-instructor/instructor-session-result-page/response-moderation-button/response-moderation-button.component.ts [hunk 3]
```diff
@@ -35,20 +34,11 @@ export class ResponseModerationButtonComponent implements OnInit {
     createdAtTimestamp: 0,
   };
 
-  @Input()
-  relatedGiverEmail: string = '';
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-session-result-page/response-moderation-button/response-moderation-button.component.ts:39]: Why is this removed?

Comment (Derek-Hardy) [reply src/web/app/pages-instructor/instructor-session-result-page/response-moderation-button/response-moderation-button.component.ts]: Overlooked its usage, have put back!

File: src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.html [hunk 4]
```diff
@@ -36,7 +36,7 @@
         [{{ userToEmail[userInfo] }}]
       </a>
       <div class="float-right">
-        <tm-response-moderation-button class="mr-md-3" *ngIf="isGqr" [session]="session" [relatedGiverEmail]="userInfo"></tm-response-moderation-button>
+        <tm-response-moderation-button class="mr-md-3" *ngIf="isGqr && userToRelatedEmail[userInfo]" [session]="session" [relatedGiverEmail]="userInfo"></tm-response-moderation-button>
```
Comment (xpdavid) [inline src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.html:39]: The relatedGiverEmail here should be `userToRelatedEmail[userInfo]`?

File: src/web/app/components/question-responses/grq-rgq-view-responses/grq-rgq-view-responses.component.html [hunk 5]
```diff
@@ -38,7 +38,7 @@
         [{{ userToEmail[userInfo] }}]
       </a>
       <div class="float-right">
-        <tm-response-moderation-button *ngIf="isGrq" [relatedGiverEmail]="userInfo"></tm-response-moderation-button>
+        <tm-response-moderation-button *ngIf="isGrq && userToRelatedEmail[userInfo]" [relatedGiverEmail]="userInfo"></tm-response-moderation-button>
```
Comment (xpdavid) [inline src/web/app/components/question-responses/grq-rgq-view-responses/grq-rgq-view-responses.component.html:41]:
```suggestion
        <tm-response-moderation-button *ngIf="isGrq && userToRelatedEmail[userInfo]" [relatedGiverEmail]="userToRelatedEmail[userInfo]"></tm-response-moderation-button>
```

File: src/web/app/pages-instructor/instructor-session-result-page/response-moderation-button/response-moderation-button.component.ts [hunk 6]
```diff
@@ -5,7 +5,6 @@ import {
   ResponseVisibleSetting,
   SessionVisibleSetting,
 } from '../../../../types/api-output';
-import { ANONYMOUS_PREFIX } from '../../../../types/feedback-response-details';
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-session-result-page/response-moderation-button/response-moderation-button.component.ts:8]: You can also remove the const I believe.
